test/vcf/compressed_file_test.cpp

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright 2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <fstream>
+
+#include "catch/catch.hpp"
+
+#include "vcf/validator.hpp"
+
+namespace ebi
+{
+
+  TEST_CASE("Tests for compressed files", "[compressed]")
+  {
+      SECTION("File with extensions of compressed files")
+      {
+          auto folder = boost::filesystem::path("test/input_files/v4.3/compressed_files");
+          std::vector<boost::filesystem::path> v;
+          copy(boost::filesystem::directory_iterator(folder), boost::filesystem::directory_iterator(), back_inserter(v));
+
+          for (auto path : v)
+          {
+              CHECK(vcf::is_compressed_extension(path.string()));
+          }
+      }
+
+      SECTION("Compressed file streams")
+      {
+          auto folder = boost::filesystem::path("test/input_files/v4.3/compressed_files");
+          std::vector<boost::filesystem::path> v;
+          copy(boost::filesystem::directory_iterator(folder), boost::filesystem::directory_iterator(), back_inserter(v));
+
+          for (auto path : v)
+          {
+              std::ifstream input{path.string()};
+              CHECK(vcf::is_compressed_magic_num(input));
```

Comment:
This are the positive cases. Please add test cases for the negative ones as well. Non compressed extensions and non-compressed filestreams, with a special check that the seekg is working as expected.

Reply:
i think `seekg()` is automatically tested in `is_valid_vcf_file` test for non compressed files in `parser_v4*` tests

Reply:
if `seekg()` behaves abnornally then for sure our `is_valid_vcf_file` test will fail for non compressedfiles passed files(as they will return `false` on `is_valid_vcf_file` ). And we will come to know something is going wrong.

Reply:
that's true, but a specific unit test is also desirable. If only the integration tests detect that, you will have dozens of them failing with strange messages, and not a single unit test failing. The point is not so much to see that current code works (which we know it does) but to protect us from future breaking changes, and being able to identify the problem quickly.